site/components/pages/Pledge/lib/firebase.ts

```diff
@@ -0,0 +1,94 @@
+import * as admin from "firebase-admin";
+import { v4 as uuidv4 } from "uuid";
+
+import { FIREBASE_ADMIN_CERT } from "lib/secrets";
+import { Footprint, Pledge, putPledgeParams } from "../types";
+import { verifySignature, DEFAULT_NONCE, generateNonce } from ".";
+
+const initFirebaseAdmin = () => {
+  if (!admin.apps.length) {
+    admin.initializeApp({
+      credential: admin.credential.cert(
+        JSON.parse(FIREBASE_ADMIN_CERT as string)
+      ),
+    });
+  }
+  return admin.firestore();
+};
+
+const buildFootprint = (
+  currentFootprint: Footprint[],
+  newFootprint: number
+): Footprint[] => {
+  if (currentFootprint.at(-1)?.total === newFootprint) return currentFootprint;
+
+  return [...currentFootprint, { timestamp: Date.now(), total: newFootprint }];
+};
+
+export const getPledgeByAddress = async (address: string): Promise<Pledge> => {
+  const db = initFirebaseAdmin();
+  const snapshot = await db
+    .collection("pledges")
+    .where("ownerAddress", "==", address)
+    .get();
+
+  const pledge = snapshot.docs[0]?.data();
+
+  return pledge as Pledge;
```

Comment:
risky, what if the given address doesn't have a pledge? This can be undefined.

I believe the exported types from firebase support generics, so you can type something like the collectionRef and then the rest should be inferred correctly. Just a suggestion 🤷 
```ts
  const pledgeCollectionRef= db.collection("pledge") as FirebaseFirestore.CollectionReference<Pledge>;
```